fv3core/stencils/mapn_tracer.py

```diff
@@ -37,10 +55,16 @@ def compute(
     # transliterated fortran 3d or 2d validate, not bit-for bit
     tracer_list = [tracers[q] for q in utils.tracer_variables[0:nq]]
     for tracer in tracer_list:
-        q4_1[:] = tracer[:]
-        q4_2[:] = 0.0
-        q4_3[:] = 0.0
-        q4_4[:] = 0.0
+        set_components(
+            tracer,
+            q4_1,
+            q4_2,
+            q4_3,
+            q4_4,
+            origin=(spec.grid.is_, spec.grid.js, 0),
```

Comment:
These changes look great to me. From the test results, it looks like the domain needs to be restricted as it is inside `remap_profile.compute`? What do you think?

Reply:
Yeah, that seems correct. Should be a pretty straightforward fix, at least.

Reply:
Nicely done!